iml-gui/crate/src/page/activity.rs

```diff
@@ -1,6 +1,262 @@
+    div![match &model.state {
+        State::Loading => loading::view(),
+        State::Loaded(alerts, rows) => div![
+            class![C.bg_menu_active],
+            div![
+                class![C.px_6, C.py_4, C.bg_blue_1000],
+                div![class![C.font_medium, C.text_lg, C.text_gray_500], "Activities"],
+                div![
+                    class![C.grid, C.grid_cols_2, C.items_center, C.text_white],
+                    div![
+                        class![C.col_span_1],
+                        format!(
+                            "Showing {} - {} of {} total",
+                            alerts.meta.offset + 1,
+                            alerts.meta.offset + alerts.meta.limit,
+                            alerts.meta.total_count
+                        )
+                    ],
+                    div![
+                        class![C.grid, C.justify_end],
+                        font_awesome(class![C.inline, C.w_4, C.h_4, C.text_green_400], "filter")
+                    ]
+                ],
+            ],
+            alerts.objects.iter().map(|x| {
+                let row = rows.get(&x.id);
+
+                alert_item_view(all_locks, session, x, row)
+            })
+        ],
+    }]
+}
+
+fn alert_icon_view(alert: &Alert) -> Node<Msg> {
+    let cls = class![C.inline, C.w_4, C.h_4];
+
+    match alert.record_type {
+        AlertRecordType::AlertEvent
+        | AlertRecordType::LearnEvent
+        | AlertRecordType::SyslogEvent
+        | AlertRecordType::StorageResourceLearnEvent => font_awesome(cls, "info-circle"),
+        x if is_cmd(x) => font_awesome(cls, "terminal"),
+        _ => font_awesome(cls, "bell"),
+    }
+}
+
+fn is_cmd(x: AlertRecordType) -> bool {
+    match x {
+        AlertRecordType::CommandRunningAlert
+        | AlertRecordType::CommandSuccessfulAlert
+        | AlertRecordType::CommandErroredAlert
+        | AlertRecordType::CommandCancelledAlert => true,
+        _ => false,
+    }
+}
+
+fn alert_item_color(alert: &Alert) -> &str {
+    match (alert.record_type, alert.severity) {
+        (AlertRecordType::CommandRunningAlert, _) => "gray",
+        (AlertRecordType::CommandSuccessfulAlert, _) => "green",
+        (AlertRecordType::CommandErroredAlert, _) => "red",
+        (AlertRecordType::CommandCancelledAlert, _) => "gray",
+        (_, AlertSeverity::DEBUG) => "gray",
+        (_, AlertSeverity::INFO) => "blue",
+        (_, AlertSeverity::WARNING) => "yellow",
+        (_, AlertSeverity::ERROR) => "red",
+        (_, AlertSeverity::CRITICAL) => "orange",
+    }
+}
+
+fn alert_item_view(all_locks: &Locks, session: Option<&Session>, alert: &Alert, row: Option<&Row>) -> Node<Msg> {
+    let color = alert_item_color(alert);
+    let border_color = format!("border-{}-500", color);
+    let icon_color = format!("text-{}-500", color);
+    let bg_color = format!("bg-{}-400", color);
```

Comment:
I am not sure these styles will be kept up by purgecss in production builds.

Reply:
good point.